### PR TITLE
Update navigation API article

### PR DESCRIPTION
**Pull request by jakearchibald**
Preview: https://deploy-preview-3390--developer-chrome-com.netlify.app/docs/web-platform/navigation-api/

- Updates the documentation and code examples for the switch from `transitionWhile` to `intercept`
- Documents scroll handling
- Documents focus handling
- More detail on the types of navigations you do/don't want to intercept
- Less focus on `updateCurrentEntry`

@domenic: mind reviewing?

@samthor: I don't think there's anything here you'd disagree with, but your name is still on this, so you might want to take a quick look.

### CODE REVIEW

**Review by samthor — COMMENTED** [review]
Looks sane to me.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:47]: Maybe move this below the early bail-out.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:61]:
```suggestion
You can deal with the navigation in one of two ways:
```

or similar?

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:73]: Doesn't seem like a correct sentence anymore.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:111]: Events in this article seem to be without code font (e.g. lots of instances of "navigate", not `"navigate"`).

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:125]:
```suggestion
Cross-origin navigations and cross-document traversals cannot be intercepted.
```

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:141]: Maybe mention that actually intercepting this is pretty common, to progressively enhance `<form>` submissions into Ajax? Forward-link to the `#form-submissions` heading?

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:119]:
```suggestion
The `navigateEvent` contains a lot of information about the navigation that you can use to decide how to deal with a particular navigation.
```

maybe? Since this is also about whether you want to cancel.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:168]:
```suggestion
When your code calls `intercept({ handler })` from within its "navigate" listener, it informs the browser that it's now preparing the page for the new, updated state, and that the navigation may take some time.
```

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:222]:
```suggestion
### Abort signals
```

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:269]: Maybe mention that this only happens if you call `intercept()`? Otherwise you get today's current behavior with no delays and such.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:452]: I appreciate the deemphasis on updateCurrentEntry(). However I hesitate a bit to leave this section without mentioning the react-to vs. initiate distinction that https://github.com/wicg/navigation-api#setting-the-current-entrys-state-without-navigating mentions. I'm not sure how to do that without expanding the section and thus perhaps overemphasizing updateCurrentEntry()... maybe just an explainer link?

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:519]: Not sure this should be changed, now it doesn't match the gif? Hmm.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (domenic) [inline site/en/docs/web-platform/navigation-api/index.md:544]: There's a leftover canTransition above this.

**Review by domenic — COMMENTED** [review]
Thanks very much!

File: site/en/docs/web-platform/navigation-api/index.md
Comment (jakearchibald) [reply site/en/docs/web-platform/navigation-api/index.md:47]: hah, yeah. I was passing it into `shouldNotIntercept` in an earlier draft.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (jakearchibald) [reply site/en/docs/web-platform/navigation-api/index.md:111]: I'll make them all use code font

File: site/en/docs/web-platform/navigation-api/index.md
Comment (jakearchibald) [reply site/en/docs/web-platform/navigation-api/index.md:141]: Shall do. Although, I wonder how many folks will do this vs handle the form as they do today, then perform a replacement navigation when the POST is complete.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (jakearchibald) [reply site/en/docs/web-platform/navigation-api/index.md:452]: I'll work it into a new section but with caveats

File: site/en/docs/web-platform/navigation-api/index.md
Comment (jakearchibald) [reply site/en/docs/web-platform/navigation-api/index.md:519]: oof, good catch, I pulled the wrong URL from my clipboard. It's supposed to be the same demo but updated for the latest API https://chief-low-specialist.glitch.me/. I'll update it.

**Comment by jakearchibald** [comment]
@rachelandrew this is ready for review and landing. The CLA check is failing due to @domenic, but he's a Googler so I presume it's fine?

**Comment by jakearchibald** [comment]
Is there anything I need to do before this can be reviewed and merged?

**Comment by rachelandrew** [comment]
@jakearchibald no - I just missed the original message (feel free to ping me on Chat if I'm ignoring something). I'll take a look now.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (rachelandrew) [inline site/en/docs/web-platform/navigation-api/index.md:69]:
```suggestion
`intercept` is available from Chrome 105.
```

File: site/en/docs/web-platform/navigation-api/index.md
Comment (rachelandrew) [inline site/en/docs/web-platform/navigation-api/index.md:73]:
```suggestion
This example calls `intercept()` on the event.
```

File: site/en/docs/web-platform/navigation-api/index.md
Comment (rachelandrew) [inline site/en/docs/web-platform/navigation-api/index.md:115]:
```suggestion
It doesn't have an analogy to `"navigate"`, except if you manually set up listeners for click events for example, as demonstrated above.
```

Comment (rachelandrew) [reply site/en/docs/web-platform/navigation-api/index.md:115]: docs style guide suggests not using e.g. or i.e.

File: site/en/docs/web-platform/navigation-api/index.md
Comment (rachelandrew) [inline site/en/docs/web-platform/navigation-api/index.md:237]:
```suggestion
This will both save the user's bandwidth, and reject the `Promise` returned by `fetch()`, preventing any following code from actions such as updating the DOM to show a now invalid page navigation.
```

**Comment by Miami87** [comment]
Rachel Andrew ***@***.***> schrieb am Do., 18. Aug. 2022,
10:58:

**Review by rachelandrew — APPROVED** [review]
Couple of small suggestions and a typo. I'll sort out the CLA override

File: site/en/docs/web-platform/navigation-api/index.md
Comment (jakearchibald) [reply site/en/docs/web-platform/navigation-api/index.md:73]: oh wow, good catch

**Comment by jakearchibald** [comment]
> I just missed the original message (feel free to ping me on Chat if I'm ignoring something). I'll take a look now.

Shall do! Didn't want to hassle you. Also, should I be pinging you directly, or is there a group I can send this stuff to?

**Comment by rachelandrew** [comment]
You can always ask in the dc.c. chat if it's just something needing a random tech writer LGTM, but if you've directly assigned something to me and I'm ignoring it (or it's urgent) just prod me on chat.

**Pull request merged**
